File: Backend/app/schemas/request.py
"""Pydantic схемы для заявок на обслуживание."""

from typing import Optional

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.contract import OwnerInfo
from app.schemas.payment import ContractInfo, PropertyInfo, TenantInfo


class RequestCreate(BaseModel):
    """Схема создания заявки."""

    contract_id: int = Field(..., gt=0)
    title: str = Field(..., min_length=3, max_length=255)
    message: str = Field(..., min_length=5, max_length=5000)
    status: str = Field(default="open", min_length=2, max_length=50)


class RequestUpdate(BaseModel):
    """Схема частичного обновления заявки."""

    title: str | None = Field(None, min_length=3, max_length=255)
    message: Optional[str] = Field(default=None, min_length=5, max_length=5000)
    status: Optional[str] = Field(default=None, min_length=2, max_length=50)


class RequestResponse(BaseModel):
    """Схема ответа по заявке."""

    id: int
    contract_id: int
    title: str
    message: str
    status: str

    model_config = ConfigDict(from_attributes=True)


class RequestWithDetailsResponse(RequestResponse):
    """Расширенный ответ с информацией о договоре, объекте и арендаторе."""

    contract_info: ContractInfo | None = None
    property_info: PropertyInfo | None = None
    tenant_info: TenantInfo | None = None  # для владельца
    owner_info: OwnerInfo | None = None  # для арендатора

File: Frontend/src/types/request.ts
// types/request.ts

export type RequestStatus = 'open' | 'in_progress' | 'completed' | 'cancelled';

export const REQUEST_STATUS_LABELS: Record<RequestStatus, string> = {
    open: 'Открыта',
    in_progress: 'В работе',
    completed: 'Выполнена',
    cancelled: 'Отменена',
};

export const REQUEST_STATUS_COLORS: Record<RequestStatus, string> = {
    open: '#ef4444',
    in_progress: '#f59e0b',
    completed: '#10b981',
    cancelled: '#6b7280',
};

export interface RequestResponse {
    id: number;
    contract_id: number;
    title: string;
    message: string;
    status: RequestStatus;
    created_at?: string;
}

export interface RequestWithDetails extends RequestResponse {
    contract_info?: {
        id: number;
        start_date: string;
        end_date: string;
        monthly_payment: string;
    };
    property_info?: {
        id: number;
        title: string;
        address: string;
    };
    tenant_info?: {
        id: number;
        name: string;
        email: string;
    };
    owner_info?: {
        id: number;
        name: string;
        email: string;
    };
}

export interface RequestCreate {
    contract_id: number;
    title: string;
    message: string;
}